crates/bevy_ecs/src/query/iter.rs

```diff
@@ -185,6 +185,41 @@ where
     }
 }
 
+/// An iterator over `K`-sized combinations of query items without repetition.
+///
+/// In this context, a combination is an unordered subset of `K` elements.
+/// The number of combinations depend on how `K` relates to the number of entities matching the [`Query`] (called `N`):
+/// - if `K = N`, only one combination exists,
+/// - if `K < N`, there are <sub>N</sub>C<sub>K</sub> combinations (see the [performance section] of `Query`),
+/// - if `K > N`, there are no combinations.
+///
+/// # Usage
+///
+/// This type is returned by calling [`Query::iter_combinations`] and related methods.
+///
+/// It can be iterated by calling [`fetch_next`](Self::fetch_next) in a `while let` loop.
+///
+/// [compiler bug]: https://github.com/rust-lang/rust/issues/62529
+/// [`Query`]: crate::system::Query
+/// [performance section]: crate::system::Query#performance
+/// [`Query::iter_combinations`]: crate::system::Query::iter_combinations
+///
+/// # Example
+///
+/// This example shows how `fetch_next` should be called with a `while let` loop to traverse the iterator.
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// # #[derive(Component)]
+/// # struct ComponentA;
+/// #
+/// fn some_system(mut query: Query<&mut ComponentA>) {
+///     let mut combinations = query.iter_combinations();
+///     while let Some([a1, a2]) = combinations.fetch_next() {
+///         // ...
+///     }
+/// }
+/// ```
```

Comment:
It cannot work with the `for` loop because it does not implement the `Iterator` trait.

The older docs in the `iter_combinations` query method specified that this type did not implement `Iterator`, but I don't think it's a good idea to describe an item in terms of what it's not.

At most I can change the first line not use the word “iterator”, as it could be misleading.

Reply:
[This comment](https://github.com/bevyengine/bevy/pull/4989/files#diff-96cc24e6d6acfeff8249b41cbf81bc003814b93a165eb49bb8edc6b14c1dc668R337-R344) explains why `Iterator` is only implemented for `QueryCombinationIter` with read-only access.

However, the documentation above does not explain this, and mistakenly uses `iter_combinations` instead of `iter_combinations_mut` in the example.

@Nilirad, do you mind clarifying these docs?

Reply:
Ooh, I completely missed that 🤣 Thanks, i'll make some changes as soon as I can.